weldx/asdf/file.py

```diff
@@ -812,6 +812,27 @@ def _ipython_display_(self):
 
         display(self.show_asdf_header(use_widgets=True, _interactive=True))
 
+    def info(
+        self, max_rows: int = None, max_length: int = None, show_values: bool = False
+    ):
+        """Print the content to the stdout.
+
+        Parameters
+        ----------
+        max_rows :
+            The maximum number of rows that will be printed. If rows are cut, a
+            corresponding message will be printed
+        max_length :
+            The maximum line length. Longer lines will be truncated
+        show_values :
+            Set to `True` if primitive values should be displayed
+
+        """
+        tree = self._asdf_handle.tree.copy()
```

Comment:
taking a copy just to remove history and asdf version seems a bit overkill, as the tree will be deeply copied. This seems prohibitive for large datasets.

Reply:
Does asdf.info also triggers serialization? If so, we should encapsulate the info call in the private Visualizer class, as this implementation does not serialize the arrays, saving resources. Eventually we could add another switch for `asdf.info` or replace the pure text representation with it.

Reply:
Since it is a function provided by asdf itself, I assume that it is not deserializing the data, but I am not sure.

Regarding the copy. I would also favor to avoid it, if there is another way to remove the dropped stuff. However, are you sure this is a deep copy and not a shallow one?

Reply:
since we are working on the 'tree' it is already de-serialized into python types (asdf itself probably doesn't *load* array data but our classes most likely will because of xarray etc.) and really just a prettier `dict` viewer imo. No asdf serialization involved imo.
For performance it would be nice to have this kind of overview generated purely form the yaml header on 'unopened' files. But that would mean working with tags instead of classes etc (which would also be nice)

regarding the copy, I guess something like the following should work by creating only a few more pointers?
```suggestion
        tree = {key:value  for key, value in self._asdf_handle.tree if key not in ["asdf_library", "history"]}
```

Reply:
So after quick research, I guess my assumption that `dict.copy` just makes a shallow copy seemed to be right. [See here](https://stackoverflow.com/questions/3975376/understanding-dict-copy-shallow-or-deep) for example. However, since @CagtayFabry suggestion does the same thing and is guaranteed to be a shallow copy, let's just take that.